pytorch_lightning/utilities/device_parser.py

```diff
@@ -50,7 +53,8 @@ def determine_root_gpu_device(gpus: List[int]) -> Optional[int]:
     return root_gpu
 
 
-def parse_gpu_ids(gpus: Optional[Union[int, str, List[int]]]) -> Optional[List[int]]:
+def parse_gpu_ids(gpus: Optional[Union[int, str, List[int]]],
+                  should_raise_exception: bool = True) -> Optional[List[int]]:
```

Comment:
nit: rename to `should_raise`?

Reply:
@kaushikb11 point was both a exception and warning are being raised. `should_raise_exception` makes it explicit.

Reply:
elsewhere in the codebase we have this as `strict` - we should be consistent on the naming

Reply:
Note that the exception/warning can be considered a side effect of the function.

It's main purpose is to parse the gpu ids.

So with that in mind, just `strict` could be confusing without looking inside the function because does it refer to "strict gpu parsing" (for example in terms of the format) or "strict gpu number checking"?